python/GafferUI/PathListingWidget.py

```diff
@@ -903,14 +936,34 @@ def updateColumnWidths( self ) :
 		self.__recalculatingColumnWidths = True
 
 		header = self.header()
-		numColumnsToResize = header.count() - 1 # leave the last section alone, as it's expandable
+		numColumnsToResize = header.count()
+		if header.stretchLastSection() or self.getStretchLastColumn() :
+			numColumnsToResize -= 1 # leave the last section alone, as it's expandable
 
+		columnWidthsChanged = False
 		for i in range( 0, numColumnsToResize ) :
-			column = self.__getColumn( i )
-			idealWidth = max( header.sectionSizeHint( i ), self.sizeHintForColumn( i ) )
-			self.__idealColumnWidths[column] = idealWidth
-
-			header.resizeSection( i, idealWidth + self.__columnWidthAdjustments[column] )
+			if self.__getColumnResizeMode( i ) == GafferUI.PathColumn.ResizeMode.Interactive :
+				column = self.__getColumn( i )
+				idealWidth = max( header.sectionSizeHint( i ), self.sizeHintForColumn( i ) )
+				self.__idealColumnWidths[column] = idealWidth
+				adjustedWidth = idealWidth + self.__columnWidthAdjustments[column]
+				if( i == header.count() - 1 ) :
```

Comment:
Do we actually need the `()` here, or is C++ just leaking into Python?

Reply:
:upside_down_face:  Addressed in c43b367.